static/app/views/performance/trends/changeExplorerUtils/spansList.tsx

```diff
@@ -0,0 +1,527 @@
+
+  const {
+    data: totalTransactionsAfter,
+    isLoading: transactionsLoadingAfter,
+    isError: transactionsErrorAfter,
+  } = useDiscoverQuery(
+    getQueryParams(
+      breakpointTime,
+      endTime,
+      ['count'],
+      'transaction',
+      DiscoverDatasets.METRICS,
+      organization,
+      trendView,
+      transaction.transaction,
+      location
+    )
+  );
+
+  const transactionCountAfter = totalTransactionsAfter?.data[0]['count()'] as number;
+
+  return (
+    <SuspectSpansQuery
+      location={beforeLocation}
+      orgSlug={organization.slug}
+      eventView={beforeEventView}
+      limit={50}
+      perSuspect={0}
+    >
+      {({
+        suspectSpans: suspectSpansBefore,
+        isLoading: spansLoadingBefore,
+        error: spansErrorBefore,
+      }) => {
+        const hasSpansErrorBefore = spansErrorBefore !== null;
+        return (
+          <SuspectSpansQuery
+            location={afterLocation}
+            orgSlug={organization.slug}
+            eventView={afterEventView}
+            limit={50}
+            perSuspect={0}
+          >
+            {({
+              suspectSpans: suspectSpansAfter,
+              isLoading: spansLoadingAfter,
+              error: spansErrorAfter,
+            }) => {
+              const hasSpansErrorAfter = spansErrorAfter !== null;
+
+              // need these averaged fields because comparing total self times may be inaccurate depending on
+              // where the breakpoint is
+              const spansAveragedAfter = addAvgSumExclusiveTime(
+                suspectSpansAfter,
+                transactionCountAfter
+              );
+              const spansAveragedBefore = addAvgSumExclusiveTime(
+                suspectSpansBefore,
+                transactionCountBefore
+              );
+
+              const addedSpans = addChangeFields(
+                findSpansNotIn(spansAveragedAfter, spansAveragedBefore),
+                true
+              );
+              const removedSpans = addChangeFields(
+                findSpansNotIn(spansAveragedBefore, spansAveragedAfter),
+                false
+              );
+
+              const remainingSpansBefore = findSpansIn(
+                spansAveragedBefore,
+                spansAveragedAfter
+              );
+              const remainingSpansAfter = findSpansIn(
+                spansAveragedAfter,
+                spansAveragedBefore
+              );
+
+              const remainingSpansWithChange = addPercentChange(
+                remainingSpansBefore,
+                remainingSpansAfter
+              );
+
+              const allSpansUpdated = remainingSpansWithChange
+                ?.concat(addedSpans ? addedSpans : [])
+                .concat(removedSpans ? removedSpans : []);
+
+              // sorts all spans in descending order of avgChange (change in avg total self time)
+              const spanList = allSpansUpdated?.sort((a, b) => b.avgChange - a.avgChange);
```

Comment:
I think this is fine for the first pass, but if possible I think a better comparison would be the average amount of time added/removed, rather than the relative difference in time. That would make added and removed spans comparable with changed spans, and you wouldn't have to hardcode avgChange values for them. For example:
* Imagine that after the breakpoint, you had one span go from 50ms to 150ms per transaction, and a new 25ms span.
* Currently, the 25ms span would come first (`avgChange: 100`, hardcoded below), followed by the 150ms span (`avgChange: 2`, since `percentChange(150, 50) === 2`)
* But, the 150ms span change is a bigger contributor to the regression (100ms vs 25ms), so ideally it would show first

Reply:
(Unrelated, but `percentChange` is a misleading function name since it doesn't actually return a percentage - you have to multiply by 100 to get the percentage)